File: 482-license-key-formatting/license-key-formatting.cpp
class Solution {
public:
    string licenseKeyFormatting(string s, int k) {
        string cleaned;
        // Step 1: Remove dashes and convert to uppercase
        for (char c : s) {
            if (c != '-') {
                cleaned += toupper(c);
            }
        }

        // Step 2: Determine size of first group
        int n = cleaned.size();
        int firstGroupSize = n % k;
        if (firstGroupSize == 0 && n > 0) firstGroupSize = k;

        string result;
        int i = 0;

        // Step 3: Add first group
        result.append(cleaned.substr(i, firstGroupSize));
        i += firstGroupSize;

        // Step 4: Add remaining groups of size k
        while (i < n) {
            result.push_back('-');
            result.append(cleaned.substr(i, k));
            i += k;
        }

        return result;
    }
};

File: 482-license-key-formatting/license-key-formatting.py
class Solution(object):
    def licenseKeyFormatting(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        # Step 1: Remove dashes and convert to uppercase
        cleaned = s.replace("-", "").upper()
        
        # Step 2: Build result from the end
        res = []
        count = 0
        
        for i in range(len(cleaned) - 1, -1, -1):
            res.append(cleaned[i])
            count += 1
            if count == k and i != 0:
                res.append('-')
                count = 0
        
        # Step 3: Reverse to get final string
        return "".join(res[::-1])
